### PR TITLE
Fix issue 4600 that search results sometimes are not displayed properly

### PR DESCRIPTION
**Pull request by ariskotsomitopoulos**
Fixing issue #4600

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/search/SearchResultController.kt [hunk 1]
```diff
@@ -54,10 +54,6 @@ class SearchResultController @Inject constructor(
         fun loadMore()
     }
 
-    init {
-        setData(null)
-    }
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/search/SearchResultController.kt:59]: I count 9 other controllers with the same code, do you think the same fix should be applied?

image

Comment (ariskotsomitopoulos) [reply vector/src/main/java/im/vector/app/features/home/room/detail/search/SearchResultController.kt:59]: I think yes, they should be removed. I will update the PR and test those screens

**Review by ouchadam — APPROVED** [review]

**Comment by bmarty** [comment]
After a discussion with @ganfra this code was added for performance reason (it should have deserve some more  comments), to not block the UI thread if the controller has many items to compute. It is maybe not necessary anymore, but  how can we be certain of that?

**Comment by ariskotsomitopoulos** [comment]
> After a discussion with @ganfra this code was added for performance reason (it should have deserve some more comments), to not block the UI thread if the controller has many items to compute. It is maybe not necessary anymore, but how can we be certain of that?

Yes that will make the `buildModels` run on background. However, this is equivalent to that #4591. We are certain that the first model build should be in UI thread otherwise the views will not be properly restored. Both the documentation and the code doc explain that. Also, this search result bug is a demonstration that this can produce problems

**Pull request merged**